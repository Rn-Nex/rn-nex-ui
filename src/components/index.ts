export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
