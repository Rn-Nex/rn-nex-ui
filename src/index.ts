export * from './components';
export * from './libraries';
export * from './utils';
