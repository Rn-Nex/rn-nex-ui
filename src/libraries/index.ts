export * from './themes/v2';
export * from './style';
