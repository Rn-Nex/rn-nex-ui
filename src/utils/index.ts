export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
