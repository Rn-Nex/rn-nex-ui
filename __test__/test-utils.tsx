import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
