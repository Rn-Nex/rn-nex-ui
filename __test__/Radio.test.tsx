import React from 'react';
import { fireEvent, render } from './test-utils';
import { Radio, Text } from '../src';
import { View } from 'react-native';

describe('Radio Component', () => {
  const mockRadioBaseButtonTestId = 'radio-base-button-test-id';

  const mockOnPress = jest.fn();
  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<Radio />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward the ref correctly', () => {
    render(<Radio ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should call the onPress function when clicked on the radio', () => {
    const { getByTestId } = render(<Radio onPress={mockOnPress} radioBaseButtonTestId={mockRadioBaseButtonTestId} />);
    const radio = getByTestId(mockRadioBaseButtonTestId);

    fireEvent.press(radio, { nativeEvent: {} });
    expect(mockOnPress).toHaveBeenCalled();
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it("should't call the onPress function when radio is disabled", () => {
    const { getByTestId } = render(<Radio disabled onPress={mockOnPress} radioBaseButtonTestId={mockRadioBaseButtonTestId} />);
    const radio = getByTestId(mockRadioBaseButtonTestId);

    fireEvent.press(radio, { nativeEvent: {} });
    expect(mockOnPress).not.toHaveBeenCalled();
    expect(mockOnPress).toHaveBeenCalledTimes(0);
  });

  it('should render the label', () => {
    const { getByText } = render(<Radio label="mock-label" />);

    const label = getByText('mock-label');
    expect(label).toBeDefined();
  });

  it('should apply the label props', () => {
    const { getByText } = render(<Radio labelProps={{ style: { color: 'red' } }} label="mock-label" />);

    const label = getByText('mock-label');
    expect(label).toBeDefined();
    expect(label.props.style).toEqual(expect.objectContaining({ color: 'red' }));
  });

  it('should render the description', () => {
    const { getByText } = render(<Radio description="mock-description" />);

    const description = getByText('mock-description');
    expect(description).toBeDefined();
  });

  it('should apply the description props', () => {
    const { getByText } = render(<Radio descriptionProps={{ style: { color: 'red' } }} description="mock-description" />);

    const description = getByText('mock-description');
    expect(description).toBeDefined();
    expect(description.props.style).toEqual(expect.objectContaining({ color: 'red' }));
  });

  it('should render the custom radio item when passed the radioItem prop', () => {
    const { getByText } = render(<Radio isActive radioItem={<Text>radioItem</Text>} />);

    const item = getByText('radioItem');
    expect(item).toBeDefined();
  });

  it('should render the adornment and not the label', () => {
    const { getByText, queryByText } = render(
      <Radio label={'mock-label'} description="mock-description" adornment={<Text>adornment</Text>} />,
    );

    const adornmentItem = getByText('adornment');
    expect(adornmentItem).toBeDefined();

    expect(queryByText('mock-label')).toBeNull();
    expect(queryByText('mock-description')).toBeNull();
  });
});
