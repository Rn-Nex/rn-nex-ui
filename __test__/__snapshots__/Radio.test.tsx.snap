// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component should render correctly 1`] = `[Function]`;
